test/web/test_server.py

```diff
@@ -43,6 +40,28 @@ def test_broken_pipe(
     assert web_session.get('/').text == ''
 
 
+@pytest.mark.xfail(reason="https://github.com/onefinestay/nameko/issues/368")
+def test_client_disconnect_os_error(
+    container_factory, web_config, web_config_port, web_session
+):
+    """ Regression for https://github.com/onefinestay/nameko/issues/368
+    """
+    container = container_factory(ExampleService, web_config)
+    container.start()
+
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.connect(('127.0.0.1', web_config_port))
+
+    with patch.object(HttpOnlyProtocol, 'handle_one_request') as handle:
+        handle.side_effect = OSError('raw readinto() returned invalid length')
```

Comment:
actually, now that we understand this, and know this is (c)python complaining about a protocol violation by eventlet, i'm happy to just simulate that exact response